src/instructlab/configuration.py

```diff
@@ -498,48 +629,105 @@ def fill_defaults(
 class _train(BaseModel):
     # model configuration
     model_config = ConfigDict(extra="ignore", protected_namespaces=())
-
-    model_path: str
-
-    data_path: str
-    ckpt_output_dir: str
-    data_output_dir: str
-
-    max_seq_len: int
-    max_batch_len: int
-    num_epochs: int
-    effective_batch_size: int
-    save_samples: int
-    checkpoint_at_epoch: bool = True
-
-    deepspeed_cpu_offload_optimizer: bool
-
-    lora_rank: int | None = None
-    lora_quantize_dtype: str | None = None
-
-    is_padding_free: bool
-
-    nproc_per_node: int
-
-    additional_args: dict[str, typing.Any]
-
+    model_path: str = Field(
+        default=DEFAULTS.MODEL_REPO,
+        description="Directory where the model to be trained is stored.",
+    )
+    data_path: str = Field(
+        default_factory=lambda: DEFAULTS.DATASETS_DIR,
+        description="Directory where datasets used for training are stored.",
```

Comment:
In the case of using the training library (primary training method now), this is a *path* to the dataset, not the directory. In the case of legacy training (MacOS/Linux), this *is* a directory.